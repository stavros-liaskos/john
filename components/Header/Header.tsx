import React from 'react';
import DarkMode from '../DarkMode/DarkMode';
import { useUser } from '@auth0/nextjs-auth0';

const Header: React.FunctionComponent = () => {
  const { user } = useUser();

  return (
    <header className="flex justify-between h-14 dark:border-slate-600 border-black border-b-2">
      <DarkMode />

      {user && (
        <button className="btn btn-small lg:ml-8 mx-3 my-2" onClick={() => (window.location.href = '/api/auth/logout')}>
          Logout
        </button>
      )}
    </header>
  );
};

export default Header;
