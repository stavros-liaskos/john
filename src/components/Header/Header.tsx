import React from 'react';
import DarkMode from '../DarkMode/DarkMode';
import Link from 'next/link';

const Header: React.FunctionComponent = () => (
  <header className="rr-column justify-center flex-none h-14 rr-border border-b-2">
    <div className="flex justify-between items-center w-full lg:w-9/12">
      <DarkMode />

      <Link href="/api/auth/logout">
        <button className="btn btn-small">Logout</button>
      </Link>
    </div>
  </header>
);
Header.whyDidYouRender = false;
export default Header;
