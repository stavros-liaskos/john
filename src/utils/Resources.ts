import { components } from '../types/schema';

interface IResources {
  cacheMap: Record<string, unknown>;
  fetch: (url: string) => void;
}

class Resources implements IResources {
  private cacheMap;

  constructor() {
    this.cacheMap = new Map();
  }

  fetch(url) {
    if (this.cacheMap.has(url)) {
      return this.cacheMap.get(url);
    }

    const promise = fetch(url, {
      method: 'GET',
      credentials: 'include',
    })
      .then(response => {
        return response.json();
      })
      .then((data: components['schemas']['FollowedArtistsResponse']) => {
        this.cacheMap.set(url, data);
        return data;
      });

    throw promise;
  }
}

const resources = new Resources();

export default resources;
