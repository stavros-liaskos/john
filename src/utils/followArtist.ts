import { components } from '../types/schema';
import Endpoints from '../types/endpoints';

async function followArtist(artist: components['schemas']['SearchResultArtistDto'], cb: () => void) {
  const headers = new Headers({ 'Content-Type': 'application/json' });
  await fetch(Endpoints.Follow, {
    method: 'POST',
    credentials: 'include',
    headers,
    body: JSON.stringify(artist),
  })
    .then(() => {
      console.log(`${artist.name} followed successfully`);
    })
    .catch(error => {
      console.error('Error:', JSON.stringify(error));
    })
    .finally(() => {
      cb();
    });
}
export default followArtist;
