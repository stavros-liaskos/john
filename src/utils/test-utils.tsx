import React from 'react';
import { render, act } from '@testing-library/react';
import ThemeProvider from '../contexts/Theme/ThemeProvider';
import { UserProvider } from '@auth0/nextjs-auth0/client';
import ArtistsListProvider from '../contexts/ArtistsList/ArtistsListProvider';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ThemeProvider>
      <UserProvider>
        <ArtistsListProvider>{children}</ArtistsListProvider>
      </UserProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: { [key: string]: unknown }) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };

export const renderWithAct = async (ui: React.ReactElement, options?: { [key: string]: unknown }) => {
  const component = render(ui, { wrapper: AllTheProviders, ...options });

  const promise = Promise.resolve();
  await act(() => promise);

  return component;
};

export const beforeEachTest = () => {
  jest.resetModules();
  jest.clearAllMocks();
  jest.restoreAllMocks();
  localStorage.clear();
};
