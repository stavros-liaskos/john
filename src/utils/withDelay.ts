export default function withDelay(cb: () => void, delay = 1000) {
  return setTimeout(() => {
    cb();
  }, delay);
}
