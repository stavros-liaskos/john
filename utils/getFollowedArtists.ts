import { components } from '../types/schema';
import { Dispatch, SetStateAction } from 'react';

export function getFollowedArtists(
  setState: Dispatch<SetStateAction<components['schemas']['FollowedArtistsResponse'] | undefined>>,
  setLoading: Dispatch<SetStateAction<boolean>>,
) {
  setLoading(true);
  const fetchData = async () => {
    const data = await fetch(`${process.env.BE_BASE_URL}/me/followed-artists`, {
      method: 'GET',
      mode: 'cors',
      headers: {
        'Content-Type': 'application/json',
      },
      referrerPolicy: 'no-referrer',
    });
    const json = await data.json();
    setState(json);
    setLoading(false);
  };
  fetchData().catch(console.error);
}
