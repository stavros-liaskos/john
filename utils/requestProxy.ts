import type { NextApiRequest, NextApiResponse } from 'next';
import headers from '../pages/api/headers';

export default async function requestProxy(req: NextApiRequest, res: NextApiResponse<unknown>) {
  const requestOptions = {
    method: req.method,
    headers,
    ...(req.body && { body: JSON.stringify(req.body) }),
  };

  try {
    const response = await fetch(process.env.BE_BASE_URL as unknown as URL, requestOptions);
    const result = await response.json();
    console.log(result);
    res.status(200).json(JSON.parse(result));
  } catch (e) {
    console.error('error', e);
  }
}
