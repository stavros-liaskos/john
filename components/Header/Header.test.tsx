import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';
import { renderWithAct } from '../../utils/test-utils';

describe('Header', () => {
  // TODO write test
  it('shows Logout btn when user is logged in', async () => {
    const { container } = await renderWithAct(<Header />);

    expect(container).toMatchSnapshot();
  });

  xit('hides Logout btn when user is logged out', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
