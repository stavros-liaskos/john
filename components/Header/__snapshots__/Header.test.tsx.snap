// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header shows Logout btn when user is logged in 1`] = `
<div>
  <header
    class="flex justify-between h-14 dark:border-slate-600 border-black border-b-2"
  >
    <button
      class="m-3"
      data-testid="moon"
    >
      <svg
        class="h-6 w-6"
        color="var(--geist-foreground)"
        fill="none"
        height="28"
        shape-rendering="geometricPrecision"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 12.79A9 9 0 1111.21 3 7 7 0 0021 12.79z"
        />
      </svg>
    </button>
  </header>
</div>
`;
